arabic-writing-practice-app/handwriting_test.py

```
@@ -0,0 +1,144 @@
+                    # Check if the image has an alpha channel (RGBA)
+                    if img_array.shape[-1] == 4:
+                        # Convert RGBA to RGB by removing the alpha channel
+                        rgb_img = img_array[:, :, :3]
+                    else:
+                        rgb_img = img_array
+                    
+                    # Create a PIL image for saving
+                    temp_img = Image.fromarray(rgb_img)
+                    
+                    # Save to a temporary file
+                    temp_file = tempfile.NamedTemporaryFile(delete=False, suffix='.png')
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)